### ISSUE TITLE
[BUG] make linking failure - cannot find -lNVCategory

### ISSUE THREAD

**Issue by prabindh** [rapidsai/cudf#4178]
**Describe the bug**
Following the build instructions for setup from source via Conda, the make step fails, due to nvstrings.

**Steps/Code to reproduce bug**
`conda env create -n cudf_dev python=3.6 --file=conda/environments/cudf_dev_cuda10.1.yml`

`cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_CXX11_ABI=ON`

Followed by make, results in below link error

```
Linking CXX shared library libcudf.so
/usr/bin/ld: cannot find -lNVCategory
collect2: error: ld returned 1 exit status
CMakeFiles/cudf.dir/build.make:3794: recipe for target 'libcudf.so' failed
```
After adding nvstrings to yml as in this patch, it builds successfully.
[Patched yml](https://github.com/prabindh/cudf/commit/edbba233f00db060955077644cfa88143929bca8)

**Expected behavior**
make should not fail

**Environment overview (please complete the following information)**
 - Bare-metal Ubuntu 18.04
 - Method of cuDF install: conda

`conda env create -n cudf_dev python=3.6 --file=conda/environments/cudf_dev_cuda10.1.yml`

**Environment details**
Please run and paste the output of the `cudf/print_env.sh` script here, to gather any other relevant environment details

Note - this is the output AFTER adding nvstrings to yml file
[link to script output](https://raw.githubusercontent.com/prabindh/deepnotes/master/rapids/post-nvstrings.txt)

**Additional context**
Python 3.7 vs Python 3.6 is not relevant to this problem, I think

**Comment by kkraus14** [comment]
@prabindh it should build NVStrings and NVCategory as part of the build, what was your `make` command after the `cmake` configuration?

**Comment by prabindh** [comment]
After the cmake step, just `make`. Exactly following the steps outlined in [https://github.com/rapidsai/cudf/blob/branch-0.13/CONTRIBUTING.md#setting-up-your-build-environment].  Without this patch, nvstrings is not installed in the environment, ie, the below line is missing in the conda list:

`nvstrings                 0.3.0           cuda9.2_py37_18    rapidsai`

**Comment by prabindh** [comment]
build.sh uses a different procedure - 
 https://github.com/rapidsai/cudf/blob/042e9d7128740743e9aac62889d672ebbe693316/build.sh#L38
https://github.com/rapidsai/cudf/blob/042e9d7128740743e9aac62889d672ebbe693316/build.sh#L39

But from CONTRIBUTING.md, it specifies build libcudf first.

https://github.com/prabindh/cudf/blob/branch-0.13/CONTRIBUTING.md#build-from-source

`Build and install libcudf`

**Comment by kkraus14** [comment]
That documentation is incorrect and NVStrings should be built before libcudf. Where you should either run just `./build.sh` or `./build.sh libnvstrings libcudf`.

Would you be interested in submitting a PR to fix the documentation?

**Comment by prabindh** [comment]
> Would you be interested in submitting a PR to fix the documentation?

Done @kkraus14  per #4208  please review

**Comment by kkraus14** [comment]
Fixed by #4208
